main.go

```diff
@@ -236,6 +238,19 @@ func authenticateHandler(c *Config) func(http.ResponseWriter, *http.Request) {
 func tokenHandler(c *Config) func(w http.ResponseWriter, r *http.Request) {
 	return func(w http.ResponseWriter, r *http.Request) {
 		username, password, ok := r.BasicAuth()
+		tokenExpiresIn := c.TokenExpiresIn
+		if v := r.URL.Query().Get("token-expires-in"); v != "" {
+			if t, err := strconv.Atoi(v); err != nil {
+				glog.Errorf("Got a non int token expiry: %s", v)
```

Comment:
Just my $0.02 but I would 400 in this case

Reply:
addressed